app/graphql/types/habit_type.rb

```diff
@@ -5,5 +5,15 @@ class HabitType < Types::BaseObject
     field :description, String, null: true
     field :created_at, GraphQL::Types::ISO8601DateTime, null: false
     field :habit_entries, [Types::HabitEntryType]
+    field :weekly_habit_entries, [Types::HabitEntryType]
+
+    def weekly_habit_entries
+      #if we add multiple users this needs to be changed, only adding it to maybe make the future easier
+      user = User.first
+      object.habit_entries
+            .where('user_id = ?', user.id)
+            .order(created_at: :desc)
+            .limit(7)
+    end
```

Comment:
Is that where clause needed?

Reply:
`user = User.find(user_id)`
`user.habit_entries.order(created_at: :desc).limit(7)`